druid/src/widget/svg.rs

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An SVG widget.
+
+use std::error::Error;
+use std::marker::PhantomData;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use log::error;
+
+use usvg;
+
+use crate::{
+    kurbo::BezPath,
+    piet::{Color, RenderContext},
+    BaseState, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, PaintCtx, Point, Size,
+    UpdateCtx, Widget,
+};
+
+/// A widget that renders a SVG
+pub struct Svg<T> {
+    svg_data: SvgData,
+    phantom: PhantomData<T>,
+}
+
+impl<T: Data> Svg<T> {
+    /// Create an SVG-drawing widget from SvgData.
+    ///
+    /// The SVG will scale to fit its box constraints.
+    pub fn new(svg_data: SvgData) -> impl Widget<T> {
+        Svg {
+            svg_data,
+            phantom: Default::default(),
+        }
+    }
+
+    /// Measure the SVG's size
+    #[allow(clippy::needless_return)]
```

Comment:
Why not use conventional style here?

Reply:
The borrow checker disagrees with clippy on the "needless" aspect: https://rust-lang.github.io/rust-clippy/master/index.html#needless_return